functions/src/project/update-repositories.ts

```diff
@@ -0,0 +1,79 @@
+// Third party modules
+import { firestore, Change, EventContext } from 'firebase-functions';
+
+// Dashboard hub firebase functions models/mappers
+import { Logger } from '../client/logger';
+import { DocumentData, DocumentSnapshot, FirebaseAdmin, WriteResult } from './../client/firebase-admin';
+
+export const onUpdateProjectRepositories: any = firestore
+  .document('projects/{projectUid}')
+  .onUpdate((change: Change<DocumentSnapshot>, context: EventContext) => {
+
+    try {
+      const newData: DocumentData = change.after.data();
+      const previousData: DocumentData = change.before.data();
+      const isArrayNewDataRepositories: boolean = Array.isArray(newData.repositories) && newData.repositories.length > 0;
+      const isArrayPreviousDataRepositories: boolean = Array.isArray(previousData.repositories) && previousData.repositories.length > 0;
+
+      let add: string[], remove: string[];
+      const promiseList: Promise<WriteResult>[] = [];
+
+      if (isArrayNewDataRepositories) {
+        if (isArrayPreviousDataRepositories) {
+          add = newData.repositories.filter((element: string) => previousData.repositories.findIndex((item: string) => element === item) === -1);
```

Comment:
Instead of `findIndex` can a `find` not be used?

Reply:
`findIndex` return accurate value `-1` when doesn`t find

Reply:
Yes I was thinking `not` find. But it's not a blocker.

```typescript
add = newData.repositories.filter((element: string) => previousData.repositories.find((item: string) => element !== item));
```